src/ex2/CompileVisitor.java

```diff
@@ -11,286 +11,329 @@
-        s = String.format("%%_%d = getelementptr [%d x i8*], [%d x i8*]* @.%s_vtable, i32 0, i32 0",
-                tmp2, classFuncs, classFuncs, e.classId());
-        addLine(s);
-        s = String.format("store i8** %%_%d, i8*** %%_%d", tmp2, tmp1);
-        addLine(s);
-        // creating the result which can be saved
-        int resReg = lastRegisterNumber++;
-        s = String.format("%%_%d = add i8* 0, %d", resReg, tmp0);
-        addLine(s);
-    }
-
-    @Override
-    public void visit(FormalArg formalArg) {
-        String s = String.format("%%%s = alloca %s", formalArg.name(), TypeDecider.llvmType(formalArg.type()));
-        addLine(s);
-    }
-
-    @Override
-    public void visit(VarDecl varDecl) {
-        String s = String.format("%%%s = alloca %s", varDecl.name(), TypeDecider.llvmType(varDecl.type()));
-        addLine(s);
-    }
-
-    // IF and JUMPS
-    @Override
-    public void visit(BlockStatement blockStatement) {
-
-    }
-
-    @Override
-    public void visit(IfStatement ifStatement) {
-
-    }
-
-    @Override
-    public void visit(WhileStatement whileStatement) {
-
-    }
-    // ARRAYS
-    @Override
-    public void visit(AssignArrayStatement assignArrayStatement) {
-    }
-
-    @Override
-    public void visit(ArrayAccessExpr e) {
-    }
-
-    @Override
-    public void visit(ArrayLengthExpr e) {
-
-    }
-
-    @Override
-    public void visit(NewIntArrayExpr e) {
-
-    }
-
-    @Override
-    public void visit(SysoutStatement sysoutStatement) {
-    }
-
-    @Override
-    public void visit(AssignStatement assignStatement) {
-        // define temp assign from rv
-    }
-
-    // Exp and trees
-    @Override
-    public void visit(AndExpr e) {
-        // implement short circuit
-    }
-
-    @Override
-    public void visit(LtExpr e) {
-
-    }
-
-    @Override
-    public void visit(AddExpr e) {
-
-    }
-
-    @Override
-    public void visit(SubtractExpr e) {
-
-    }
-
-    @Override
-    public void visit(MultExpr e) {
-
-    }
-
-    @Override
-    public void visit(NotExpr e) {
-
-    }
-
-    // SIMPLE
-    @Override
-    public void visit(IntegerLiteralExpr e) {
-
-    }
-
-    @Override
-    public void visit(TrueExpr e) {
-
-    }
-
-    @Override
-    public void visit(FalseExpr e) {
-
-    }
-
-    @Override
-    public void visit(IdentifierExpr e) {
-
-    }
-
-
-    @Override
-    public void visit(IntAstType t) {
-
-    }
-
-    @Override
-    public void visit(BoolAstType t) {
-
-    }
-
-    @Override
-    public void visit(IntArrayAstType t) {
-
-    }
-
-    @Override
-    public void visit(RefType t) {
-
-    }
+	private final StringBuilder builder = new StringBuilder();
+	private final ClassMapping classMapping;
+	private Map<Symbol, Integer> symbolMapping = null; // maps symbol to its index, no need to offset fields
+	private Map<String, ClassPointer> pointerMap = null;
+	private int lastRegisterNumber = 0;
+	private int lastLabelNumber = 0;
+	private String expr;
```

Comment:
Using this string to remember last value is not a good idea, instead you should load the value into a register in the final line

Reply:
how can I load integer into a register?